### PR TITLE
Fix self-closing tags, add React tests

### PR DESCRIPTION
**Pull request by sigorilla**
Ref: #1, #2

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -34,7 +36,12 @@ function renderElement(element, options) {
             content = renderChildren([].concat(props.children), options);
         }
 
-        return '<' + type + renderAttrs(props) + '>' + content + '</' + type + '>';
+        var attrs = renderAttrs(props);
```
Comment (alt-j) [inline src/index.js:39]: what benchmarks says about this fix?

File: src/consts.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+    selfClosingTags: ['area', 'base', 'br', 'col', 'command', 'embed', 'hr', 'img', 'input',
```
Comment (alt-j) [inline src/consts.js]: i think, we can put in index.js

File: test/index.js [hunk 3]
```diff
@@ -1,11 +1,212 @@
...
+        it('should render tag correctly', function () {
+            var element = React.createElement('div');
+            var expectString = '<div></div>';
+            expect(ReactRender.elementToString(element)).to.equal(expectString);
+        });
+
+        it('should render tag with attributes correctly', function () {
+            var element = React.createElement('a', {
+                href: '//ya.ru',
+                className: 'link'
```
Comment (alt-j) [inline test/index.js]: may be add test for different types of attributes?

File: test/index.js [hunk 4]
```diff
@@ -1,11 +1,212 @@
...
+            var element = React.createElement('input');
+            var expectString = '<input />';
+            expect(ReactRender.elementToString(element)).to.equal(expectString);
+        });
+
+        it('should render self-closing tag with attributes correctly', function () {
+            var element = React.createElement('input', {
+                type: 'password',
+                value: 'pass',
+                readOnly: true
```
Comment (alt-j) [inline test/index.js:74]: may be the same one, but with `false` value?

File: test/index.js [hunk 5]
```diff
@@ -1,11 +1,212 @@
...
+                    return React.createElement(
+                        'span',
+                        {className: 'name'},
+                        'Name: ' + this.context.name
+                    );
+                }
+            });
+            var element = React.createElement(Context, null, React.createElement(Component));
+            var expectString = '<div><span class="name">Name: Aeron</span></div>';
+            expect(ReactRender.elementToString(element)).to.equal(expectString);
```
Comment (alt-j) [inline test/index.js:215]: I think, we need one more test with call like this `ReactRender.elementToString(element, {context: {some data}})`

**Comment by alt-j** [comment]
And if we add tests for cache, we can return back badge with coverage :)

File: src/index.js (hunk 1 above)
Comment (sigorilla) [reply src/index.js:39]: I don't see any increasing of benchmark time.

Comment (alt-j) [reply src/index.js:39]: 👍

File: test/index.js [hunk 6]
```diff
@@ -229,5 +230,73 @@ describe('ReactRender', function () {
...
+            var Component = React.createClass({
+                displayName: 'Component',
+
+                getDefaultProps: function () {
+                    return {
+                        content: content
+                    };
+                },
+
+                getCacheKey: function () {
```
Comment (alt-j) [inline test/index.js:249]: maybe add case: component hasn't getCacheKey key function

**Comment by alt-j** [comment]
Great one!

Squash your commits, please!

**Comment by sigorilla** [comment]
Yep, I also added coverage badge 😸 

FYI: Github [allow](https://github.com/blog/2141-squash-your-commits) squash PR

**Pull request merged**